frappe/public/js/frappe/list/list_view.js

```diff
@@ -1358,6 +1365,7 @@ frappe.views.ListView = class ListView extends frappe.views.BaseList {
 		if (!cur_list || route[0] != "List" || cur_list.doctype != route[1]) {
 			// wait till user is back on list view before refreshing
 			this.pending_document_refreshes = [];
+			this.disable_realtime_updates();
```

Comment:
This is kinda intentional to not unsubscribe immediately but only unsubscribe when next event is sent. 

If we unsubscribe immediately each navigation back and forth will keep sending request to subscribe and unsubscribe. Many list views aren't that active so we don't need to worry about unsubscribing.